### ISSUE TITLE
`About` reports reduction behavior incorrectly

### ISSUE THREAD

**Issue by Janno** [coq/coq#13646]
#### Description of the problem
```
Definition pred (n : nat) :=
  match n with
  | O => O
  | S n => n
  end.
Arguments pred !_.
About pred.
```
outputs
```
The reduction tactics unfold pred when the 1st argument evaluates to a constructor
```
when it should really say
```
The reduction tactics does not unfold pred unless the 1st argument evaluates to a constructor
```

https://github.com/coq/coq/issues/4555#issuecomment-337539969 indicates that this is the actual semantics here.

#### Coq Version
8.12.1; most likely all recent version.

**Comment by Janno** [comment]
I tried fixing it myself but I don't understand the code in https://github.com/coq/coq/blob/6c35e25cfa8fd47a7aa856de005e7778b5ca0e21/pretyping/reductionops.ml#L137-L169 
I suspect it will be easy to fix this for somebody who knows the different cases here.

**Comment by SkySkimmer** [comment]
>when it should really say

Isn't it the same thing with different words?

**Comment by Janno** [comment]
You are right. My proposal is flawed. It's not easy for me to find the correct description. The comment I linked to has this here:
> The ! in Arguments does not "enable simplification when the last argument is a constructor". Rather, it prevents simplification when the last argument is not a constructor.

How would one best describe this in the output of `About`?

**Comment by Blaisorblade** [comment]
The correct issue is that "that arg is a constructor" is _necessary but not sufficient_ for reduction, as #4555 points out. 

That gives
> The reduction tactics do not unfold pred if the 1st argument does not evaluate to a constructor

but I'd consider

> To unfold pred, it's necessary but not sufficient that the 1st argument evaluates to a constructor.

**Comment by ppedrot** [comment]
Why do we need to put this in words when printing the arguments flag would be much less error-prone and much more robust?

**Comment by Janno** [comment]
> Why do we need to put this in words when printing the arguments flag would be much less error-prone and much more robust?

Works for me. The documentation seems to get this particular thing right (consistently, in several places!) so users would have to look it up but they already have to do that now because the output is wrong.

**Comment by Alizter** [comment]
In 8.15+alpha we get this output:
```coq
pred : nat -> nat

pred is not universe polymorphic
Arguments pred !n%nat_scope
The reduction tactics unfold pred
  when the 1st argument evaluates to a constructor
pred is transparent
Expands to: Constant Top.pred
```

Are we suggesting removing the info about reduction tactics completely since `Arguments` already contains that info?
